terraform/paas/backend.tf

```diff
@@ -0,0 +1,6 @@
+terraform {
+  backend "azurerm" {
+    container_name = "paas-tfstate"
+    key            = "paas-qa-find.tfstate"
```

Comment:
This can't have hard coded environment here as it is the same code for multiple environments.
I suggest to use just "paas-find.tfstate"